src/sagas/featureToggle.js

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Platform } from 'react-native';
+import { UnleashClient, EVENTS as UnleashEvents } from 'unleash-proxy-client';
+import { get } from 'lodash';
+
+import {
+  takeEvery,
+  all,
+  call,
+  delay,
+  put,
+  cancelled,
+  select,
+  race,
+  take,
+  fork,
+  spawn,
+} from 'redux-saga/effects';
+import { eventChannel } from 'redux-saga';
+import { getUniqueId } from 'react-native-device-info';
+import {
+  setUnleashClient,
+  setFeatureToggles,
+  featureToggleInitialized,
+} from '../actions';
+import {
+  UNLEASH_URL,
+  UNLEASH_CLIENT_KEY,
+  UNLEASH_POLLING_INTERVAL,
+  STAGE,
+  FEATURE_TOGGLE_DEFAULTS,
+} from '../constants';
+
+const CONNECT_TIMEOUT = 10000;
+const MAX_RETRIES = 5;
+
+export function* handleInitFailed(currentRetry) {
+  if (currentRetry >= MAX_RETRIES) {
+    console.error('Max retries reached while trying to create the unleash-proxy client.');
```

Comment:
The only problem I see with using Sentry is that we should send information to it only if the user allows it, and I don't think would be a good UX to show the alert error here. I feel keeping the default for each feature flag is the best approach because of that

Reply:
IMO we should just continue with the defaults, otherwise we would spam our users with error prompts if for some reason our unleash server is down